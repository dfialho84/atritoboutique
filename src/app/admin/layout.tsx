import AdminNavbar from "@/components/admin/AdminNavbar";
import Footer from "@/components/Footer";
import BreadcrumbProvider from "@/contexts/BreadcrumbProvider";

export default function AdminLayout({
    children,
}: Readonly<{
    children: React.ReactNode;
}>) {
    return (
        <BreadcrumbProvider>
            <div className="flex flex-col min-h-screen">
                <AdminNavbar />
                <div className="flex-1 container mx-auto">{children}</div>
                <Footer />
            </div>
        </BreadcrumbProvider>
    );
}
