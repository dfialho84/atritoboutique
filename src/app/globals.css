@import "tailwindcss";

:root {
    --background: #ffffff;
    --foreground: #171717;
}

@font-face {
    font-family: "Andes";
    src: url("/fonts/Andes.ttf") format("truetype");
    font-weight: normal;
    font-style: normal;
}

@theme inline {
    --color-background: var(--background);
    --color-foreground: var(--foreground);
    --font-sans: var(--font-geist-sans);
    --font-mono: var(--font-geist-mono);
}

@theme {
    --color-primary: #8b5cf6;
    --font-andes: "Andes", sans-serif;
}

@media (prefers-color-scheme: light) {
    :root {
        --background: #0a0a0a;
        --foreground: #ededed;
    }
}

body {
    background: var(--background);
    color: var(--foreground);
    font-family: Arial, Helvetica, sans-serif;
}
