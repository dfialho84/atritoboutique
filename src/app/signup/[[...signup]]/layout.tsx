export default function SignUpLayout({
    children,
}: {
    children: React.ReactNode;
}) {
    return (
        <div className="min-h-screen bg-white text-neutral-900">{children}</div>
    );
}
