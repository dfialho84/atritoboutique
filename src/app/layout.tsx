import type { Metadata } from "next";
import { ClerkProvider } from "@clerk/nextjs";
import { ptBR } from "@clerk/localizations";
import { Toaster } from "sonner";
import UrlToaster from "@/components/UrlToaster";
import "./globals.css";

export const metadata: Metadata = {
    title: "Atrito Boutique",
    description: "Loja de roupas e acess√≥rios",
};

export default function RootLayout({
    children,
}: Readonly<{
    children: React.ReactNode;
}>) {
    return (
        <html lang="pt-BR">
            <body>
                <ClerkProvider
                    localization={ptBR}
                    afterSignOutUrl="/"
                    signInUrl="/signup"
                    signUpForceRedirectUrl="/"
                    signInForceRedirectUrl="/"
                >
                    <main className="min-h-screen bg-white text-neutral-900">
                        {children}
                    </main>
                    <Toaster richColors position="bottom-center" />
                    <UrlToaster />
                </ClerkProvider>
            </body>
        </html>
    );
}
