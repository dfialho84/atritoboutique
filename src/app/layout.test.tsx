import { render } from "@testing-library/react";
import RootLayout from "./layout";
import { expect } from "@jest/globals";
import { metadata } from "./layout";

import { screen } from "@testing-library/react";

jest.mock("@clerk/nextjs", () => ({
    ClerkProvider: ({
        children,
    }: Readonly<{
        children: React.ReactNode;
    }>) => <>{children}</>,
}));

jest.mock("sonner", () => ({
    Toaster: jest.fn(),
}));

jest.mock("../components/UrlToaster", () => jest.fn());

describe("Layout", () => {
    it("renders the layout with children", () => {
        render(
            <RootLayout>
                <div>Test Content</div>
            </RootLayout>
        );
        expect(screen.getByText("Test Content")).toBeInTheDocument();
    });

    it("should match the snapshot", () => {
        const { container } = render(
            <RootLayout>
                <div>Test Content</div>
            </RootLayout>
        );
        expect(container).toMatchSnapshot();
    });
});

describe("Metadata", () => {
    it("has the correct title and description", () => {
        expect(metadata).toEqual({
            title: "Atrito Boutique",
            description: "Loja de roupas e acess√≥rios",
        });
    });
});
