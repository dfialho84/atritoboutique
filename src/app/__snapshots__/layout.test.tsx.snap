// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Layout should match the snapshot 1`] = `
<div>
  <main
    class="min-h-screen bg-white text-neutral-900"
  >
    <div>
      Test Content
    </div>
  </main>
</div>
`;
